expression/builtin_compare.go

```diff
@@ -847,6 +847,62 @@ func getCmpType(lhs, rhs types.EvalType, lft, rft *types.FieldType) types.EvalTy
 	return types.ETReal
 }
 
+func getCmpType(lhs, rhs Expression, moreAccure bool) types.EvalType {
```

Comment:
how about set `getCmpType` to be a member function of `compareFunctionClass` ?

Reply:
`in` will use this to check whether it has no need to be rewritten to `or`.